src/js/components/componentDiagram/graph/util.js

```diff
@@ -0,0 +1,97 @@
+export function generateHash() {
+  return Math.random().toString(16).slice(2, 10);
+}
+
+export function getAnimationStep(from, to, percent) {
+  return percent < 1 ? from + (to - from) * percent : to;
+}
+
+export function createSVGElement(tagName, className = null) {
+  const el = document.createElementNS('http://www.w3.org/2000/svg', tagName);
+
+  if (className) {
+    el.setAttribute('class', className);
+  }
+
+  return el;
+}
+
+export function transformPoints(points) {
+  const result = [];
+
+  points.forEach((p) => {
+    result.push([p.x, p.y]);
+  });
+
+  return result;
+}
+
+export function normalizePoints(points1, points2) {
```

Comment:
Can you add a comment to explain what this is? To me, the signature reads like some sort of vector normalization.

Reply:
Added the comment: 
`when we need to transition between 2 edges with different points length - fill smaller array with copies of it's first point`